api/src/Controller/JWTController.php

```diff
@@ -32,4 +37,12 @@ public function getPublicJwkKey(): JsonResponse
 
         return new JsonResponse($jwk);
     }
+
+    /**
+     * @Route("/jwt/authorise", methods={"GET"})
+     */
+    public function checkJWTValid(Request $request): bool
```

Comment:
Not sure why the validity checking is part of this controller and requiring an HTTP request to invoke. Would expect it to be either a base class that the jwt-authorised endpoints extend, a trait or some middleware. Part of the reason to use jwt is you can validate a token without an external request.

Reply:
Yeah it was to avoid duplicating the code on the client side. So we have the verification on the API side but all of the document sync process on the client side. I wasn't sure how I would validate the JWT otherwise without rewriting all of the code to be API side or duplicating the validation to the client side (as we still want to be able to auth API side).

If we were to have client side API auth with JWT, how should that be processed then? What would be your thoughts on a separate JWT auth service eventually that sat outside the front and api containers? Is that a legit thing to do?

Reply:
Couldn't quite bring myself to completely rewrite everything, not least as I was going for a 2 stage approach of releasing this out with the rest of the config in place, then toggling it to run for a while in prod to bed in before removing all the original task spin ups. I have duplicated a couple of the backend functions (to be removed in cleanup PR) so all the auth logic is done in backend only and we're only passing the token through (as that was your main concern). Hopefully that's ok :)